shared/src/service/transport.ts

```diff
@@ -36,6 +36,7 @@ export interface ApiResponse<T> {
 export interface TransportService {
   /** Submit the Google API request and resolve its response. */
   submit<T>(request: ApiRequest): Promise<ApiResponse<T>>;
+  setAccessToken?(accessToken: string): Promise<void>;
```

Comment:
Since this is only applicable to the ClientTransportService, I would suggest that you don't make it part of the general TransportService interface and instead add it specifically to the ClientTransportService. Then, in the NotebookService, you can explicitly declare the constructor as requiring a ClientTransportService object.

Reply:
Mainly had it here for testing purposes - I'll change it!